### PR TITLE
Further SCIM sync testing, minor fixes

### PR DESCRIPTION
**Pull request by Firstyear**
This adds further testing of SCIM sync, especially around conversion of the SCIM Sync Person and Group types into SCIM Entry. This test would have prevented #3298 and #3299 from occurring.

During testing two more fixes were found. external_id should have been required (not optional) and a group with no members would cause a serialisation issue.

Checklist

- [x] This PR contains no AI generated code
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: proto/src/scim_v1/synch.rs [hunk 1]
```diff
@@ -128,7 +133,7 @@ impl ScimSyncPerson {
                         SCIM_SCHEMA_SYNC_PERSON.to_string(),
                     ],
                     id,
-                    external_id: None,
+                    external_id: Some(external_id),
```
Comment (CEbbinghaus) [inline proto/src/scim_v1/synch.rs:136]: Not 100% certain on whether we expect the `ScimSyncPerson` to be sent to us without `external_id` (or really anything scim related for that matter), But if not do we need the `external_id` field to still be `Option<String>`?

**Review by CEbbinghaus — APPROVED** [review]
Just a question. Looks nice otherwise

File: proto/src/scim_v1/synch.rs (hunk 1 above)
Comment (Firstyear) [reply proto/src/scim_v1/synch.rs:136]: SCIM rfc states that external ID is optional, but in our usage of it, it's required. That's why there is a disparity here.

Comment (CEbbinghaus) [reply proto/src/scim_v1/synch.rs:136]: No stress, Just thought to ask because I wasn't sure. Good to go

**Review by yaleman — APPROVED** [review]
More testing more better nice work 🥳

**Pull request merged**